### PR TITLE
Fixes rewards disabled for ads

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/2922

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
- enable rewards
- go to settings page
- disable rewards
- make sure that you don't see toggle or settings page in ads box

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [x] Verify test plan is specified in PR before merging to source

### CODE REVIEW

**Comment by jasonrsadler** [comment]
Should test plan be `disable rewards`?

**Comment by NejcZdovc** [comment]
@jasonrsadler you are correct, fixed

File: browser/ui/webui/brave_rewards_ui.cc [hunk 1]
```diff
@@ -804,19 +804,11 @@ void RewardsDOMHandler::GetAdsData(const base::ListValue *args) {
   if (ads_service_ && web_ui()->CanCallJavascript()) {
     base::DictionaryValue adsData;
 
-    bool ads_ui_enabled;
     bool ads_enabled = ads_service_->is_enabled();
     int ads_per_hour = ads_service_->ads_per_hour();
 
-    #if BUILDFLAG(BRAVE_ADS_ENABLED)
```
Comment (bridiver) [inline browser/ui/webui/brave_rewards_ui.cc]: is it now disabled somewhere else based on the buildflag?

Comment (NejcZdovc) [reply browser/ui/webui/brave_rewards_ui.cc]: @bridiver not that I am aware of

Comment (NejcZdovc) [reply browser/ui/webui/brave_rewards_ui.cc]: cc @tmancey

**Pull request merged**